### PR TITLE
feat(autoware_lidar_centerpoint): shuffled points before feeding them to the model

### PR DESCRIPTION
**Pull request by knzo25**
## Description
shuffled points before feeding them to the model to avoid voxels filling with highly correlated points

Inference time:
After this PR: 9.69ms
Before this PR: 8.49 ms

## Related links

**Parent Issue:**

- Link

Related thread
- [TIER IV](https://star4.slack.com/archives/C03S84LDJGG/p1725614367350419)

## How was this PR tested?
Used the X2 data from the thread and a reference bag from XX1.
Tested both centerpoint (the model from x2 and the standard model) and transfusion

## Notes for reviewers

None.

## Interface changes

None.

## Effects on system behavior

None.

### CODE REVIEW

File: perception/autoware_lidar_centerpoint/src/node.cpp [hunk 1]
```diff
@@ -104,10 +104,12 @@ LidarCenterPointNode::LidarCenterPointNode(const rclcpp::NodeOptions & node_opti
       rclcpp::get_logger("lidar_centerpoint"),
       "The size of voxel_size != 3: use the default parameters.");
   }
+
```
Comment (kminoda) [inline perception/autoware_lidar_centerpoint/src/node.cpp:107]: Would you remove this irrelevant change?
```suggestion
```

File: perception/autoware_lidar_centerpoint/src/node.cpp [hunk 2]
```diff
@@ -104,10 +104,12 @@ LidarCenterPointNode::LidarCenterPointNode(const rclcpp::NodeOptions & node_opti
...
   CenterPointConfig config(
     class_names_.size(), point_feature_size, cloud_capacity, max_voxel_size, point_cloud_range,
     voxel_size, downsample_factor, encoder_in_feature_size, score_threshold,
     circle_nms_dist_threshold, yaw_norm_thresholds, has_variance_);
+
```
Comment (kminoda) [inline perception/autoware_lidar_centerpoint/src/node.cpp:112]: Would you remove this irrelevant change?
```suggestion
```

File: perception/autoware_lidar_centerpoint/lib/preprocess/voxel_generator.cpp [hunk 3]
```diff
@@ -51,7 +51,7 @@ bool VoxelGeneratorTemplate::enqueuePointCloud(
 
 std::size_t VoxelGenerator::generateSweepPoints(float * points_d, cudaStream_t stream)
 {
-  size_t point_counter = 0;
+  std::size_t point_counter = 0;
```
Comment (kminoda) [inline perception/autoware_lidar_centerpoint/lib/preprocess/voxel_generator.cpp:54]: Since this change is irrelevant from the core idea of this PR which may confuse other developers who will refer to this PR in the future, can we merge the following PR first?
https://github.com/autowarefoundation/autoware.universe/pull/8820

File: perception/autoware_lidar_centerpoint/include/autoware/lidar_centerpoint/centerpoint_config.hpp [hunk 4]
```diff
@@ -116,7 +116,7 @@ class CenterPointConfig
   std::size_t head_out_rot_size_{2};
   std::size_t head_out_vel_size_{2};
 
-  // post-process params
+  // post-processing params
```
Comment (kminoda) [inline perception/autoware_lidar_centerpoint/include/autoware/lidar_centerpoint/centerpoint_config.hpp:119]: Is this change necessary?
```suggestion
  // post-process params
```

File: perception/autoware_lidar_centerpoint/lib/centerpoint_trt.cpp [hunk 5]
```diff
@@ -135,7 +154,13 @@ bool CenterPointTRT::preprocess(
   if (!is_success) {
     return false;
   }
-  const auto count = vg_ptr_->generateSweepPoints(points_d_.get(), stream_);
+
+  const std::size_t count = vg_ptr_->generateSweepPoints(points_aux_d_.get(), stream_);
+  const std::size_t random_offset = std::rand() % config_.cloud_capacity_;
+  shufflePoints_launch(
+    points_aux_d_.get(), shuffle_indices_d_.get(), points_d_.get(), count, config_.cloud_capacity_,
+    random_offset, stream_);
```
Comment (kminoda) [inline perception/autoware_lidar_centerpoint/lib/centerpoint_trt.cpp:162]: Any specific reasons for not adding `CHECK_CUDA_ERROR`? (Sorry if I misunderstand CHECK_CUDA_ERROR)
```suggestion
  CHECK_CUDA_ERROR(shufflePoints_launch(
    points_aux_d_.get(), shuffle_indices_d_.get(), points_d_.get(), count, config_.cloud_capacity_,
    random_offset, stream_));
```

File: perception/autoware_lidar_centerpoint/include/autoware/lidar_centerpoint/centerpoint_config.hpp (hunk 4 above)
Comment (knzo25) [reply perception/autoware_lidar_centerpoint/include/autoware/lidar_centerpoint/centerpoint_config.hpp:119]: Other than grammar, no

File: perception/autoware_lidar_centerpoint/lib/preprocess/voxel_generator.cpp (hunk 3 above)
Comment (kminoda) [reply perception/autoware_lidar_centerpoint/lib/preprocess/voxel_generator.cpp:54]: Sorry I missed some `size_t` in the above PR :cry: 
Let's just merge this PR as it is.

File: perception/autoware_lidar_centerpoint/include/autoware/lidar_centerpoint/centerpoint_config.hpp (hunk 4 above)
Comment (kminoda) [reply perception/autoware_lidar_centerpoint/include/autoware/lidar_centerpoint/centerpoint_config.hpp:119]: This is just a nits, but can we revert this change?

**Review by kminoda — APPROVED** [review]
